pkg/account/api/search_filter.go

```diff
@@ -174,3 +174,73 @@ func (s *AccountService) getUpdateSearchFilterCommands(req *accountproto.UpdateS
 	}
 	return commands
 }
+
+func (s *AccountService) DeleteSearchFilter(
+	ctx context.Context,
+	req *accountproto.DeleteSearchFilterRequest,
+) (*accountproto.DeleteSearchFilterResponse, error) {
+	localizer := locale.NewLocalizer(ctx)
+	editor, err := s.checkEnvironmentRole(
+		ctx, accountproto.AccountV2_Role_Environment_VIEWER,
+		req.EnvironmentId, localizer)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := validateDeleteSearchFilterRequest(req, localizer); err != nil {
+		s.logger.Error(
+			"Failed to delete search filter",
+			log.FieldsFromImcomingContext(ctx).AddFields(
+				zap.Error(err),
+			)...,
+		)
+		return nil, err
+	}
+
+	if err := s.updateAccountV2MySQL(
+		ctx,
+		editor,
+		[]command.Command{req.Command},
+		req.Email,
+		req.OrganizationId); err != nil {
+		if errors.Is(err, v2as.ErrAccountNotFound) || errors.Is(err, v2as.ErrAccountUnexpectedAffectedRows) {
+			dt, err := statusNotFound.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalizeWithTemplate(locale.NotFoundError),
+			})
+			if err != nil {
+				return nil, statusInternal.Err()
+			}
+			return nil, dt.Err()
+		}
+		if errors.Is(err, domain.ErrSearchFilterNotFound) {
+			dt, err := statusSearchFilterIDNotFound.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalizeWithTemplate(locale.NotFoundError),
+			})
+			if err != nil {
+				return nil, statusInternal.Err()
+			}
+			return nil, dt.Err()
+		}
+		s.logger.Error(
```

Comment:
The log must be at the top. Otherwise, it won't print the log when the account or the filter is not found.

Reply:
> The log must be at the top. Otherwise, it won't print the log when the account or the filter is not found.

I moved it because there was a problem with the call location in other APIs as well.
https://github.com/bucketeer-io/bucketeer/pull/1227/commits/459f4d9dfe5a63c44cae3d57da750f48a2081acc